File: dbfit-2.x/dbfit-core/src/main/java/dbfit/util/Options.java
package dbfit.util;

import java.util.HashMap;
import java.util.Map;

public class Options {
	public static void reset() {
		fixedLengthStringParsing = false;
		bindSymbols = true;
		debugLog = false;
	}

	private static boolean fixedLengthStringParsing = false;
	private static boolean bindSymbols = true;
	private static boolean debugLog = false;

	public static boolean isFixedLengthStringParsing() {
		return fixedLengthStringParsing;
	}

	public static boolean isBindSymbols() {
		return bindSymbols;
	}

	public static boolean isDebugLog() {
		return debugLog;
	}
	private static Map<String,String> freeOptions=new HashMap<String,String>();
	
	public static boolean is(String option){
		String normalname = NameNormaliser.normaliseName(option);
		if (!freeOptions.containsKey(normalname)) return false;
		return Boolean.parseBoolean(freeOptions.get(normalname));
	}
	public static void setOption(String name, String value) {
		String normalname = NameNormaliser.normaliseName(name);
		if ("fixedlengthstringparsing".equals(normalname)) {
			fixedLengthStringParsing = Boolean.parseBoolean(value);
		} else if ("bindsymbols".equals(normalname)) {
			bindSymbols = Boolean.parseBoolean(value);
		} else if ("debuglog".equals(normalname)) {
			debugLog = Boolean.parseBoolean(value);
		} else
			freeOptions.put(normalname,value);
	}
}

File: dbfit-dotnet/core/src/util/Options.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace dbfit.util
{
    public class Options
    {
        public static void reset()
        {
            fixedLengthStringParsing = false;
            bindSymbols = true;
            RemoveHandler("FailKeywordHandler");
            LoadHandler("dbfit.util.RedirectingFailHandler");
        }
        private static bool fixedLengthStringParsing = false;
        public static bool IsFixedLengthStringParsing()
        {
            return fixedLengthStringParsing;
        }
        private static bool bindSymbols = true;
        public static bool ShouldBindSymbols()
        {
            return bindSymbols;
        }

        public static void SetOption(String name, String value)
        {
            String normalname = NameNormaliser.NormaliseName(name);
            if ("fixedlengthstringparsing".Equals(normalname))
            {
                fixedLengthStringParsing = Boolean.Parse(value);
                if (fixedLengthStringParsing)
                    LoadHandler("dbfit.util.FixedLengthStringHandler");
                else
                    RemoveHandler("dbfit.util.FixedLengthStringHandler");
            }
            else if ("bindsymbols".Equals(normalname))
            {
                bindSymbols = Boolean.Parse(value);
            }
            else throw new ApplicationException("Unsupported option" + name);
        }
        private static void LoadHandler(String handler)
        {
            ((fit.CellHandlerList)fit.Configuration.Instance[fit.Configuration.FitCellHandlersKey]).
                Add(handler);
        }
        private static void RemoveHandler(String handler)
        {
            ((fit.CellHandlerList)fit.Configuration.Instance[fit.Configuration.FitCellHandlersKey]).
                Remove(handler);
        }
    }
}
